compactHeadersApi.js

```diff
@@ -190,6 +190,67 @@ function install(window) {
     msgHeaderView.append(compactHeadersPopup);
   }
 
+  function patchRecipientClass() {
+    window.customElements.whenDefined("header-recipient").then(classHeaderRecipient => {
+      if (!classHeaderRecipient.prototype.originalUpdateRecipient) {
+        classHeaderRecipient.prototype.originalUpdateRecipient = classHeaderRecipient.prototype.updateRecipient;
+        classHeaderRecipient.prototype.updateRecipient = function() {
+          this.originalUpdateRecipient();
+
+          if (this.dataset.headerName == "to" || this.dataset.headerName == "cc") {
+            this.nameLine.textContent = this.displayName;
+            this.addressLine.textContent = this.emailAddress;
+            if (this.displayName) {
+              this.classList.add("has-display-name");
+            } else {
+              this.classList.remove("has-display-name");
+            }
+          }
+        }
+        // Call updateRecipient for existing recipients before patched
+        for (let recipient of expandedtoBox.querySelectorAll('li[is="header-recipient"]')) {
+          recipient.updateRecipient();
+        }
+        for (let recipient of expandedccBox.querySelectorAll('li[is="header-recipient"]')) {
+          recipient.updateRecipient();
+        }
+      }
+    });
+  }
+
+  function createStyle() {
+    if (!document.getElementById("compactHeadersStyle")) {
+      let style = document.createElement('style');
+      style.id = "compactHeadersStyle";
+      style.textContent = `
+#messageHeader[compact="compact"] :is(#headerSenderToolbarContainer, #expandedtoRow, #expandedccRow) {
+  min-width: 0;
+}
+
+#messageHeader[compact="compact"] :is(#expandedtoRow, #expandedccRow) .recipients-list {
+  flex-wrap: nowrap;
+}
+
+#messageHeader[compact="compact"] :is(#expandedtoRow, #expandedccRow) .recipients-list .header-recipient {
+  min-width: max-content;
+}
```

Comment:
By using `min-width: max-content` for each recipient, we avoid recipient full address spanming across multiple lines when space is not suffcient:
![](https://github.com/dillenger/compact-headers/assets/20227484/08be0905-26f3-4b6d-824f-15144c68c66d)
(Each recipient single-line name+address is wrapped to three lines, with `min-width: max-content`, they stick to single line)